app/src/main/java/com/ubergeek42/WeechatAndroid/relay/Lines.kt

```diff
@@ -87,13 +93,39 @@ class Lines {
     // in very rare cases status might not be FETCHING here, particularly when closing buffers
     // while the app is connecting and has already requested lines
 
-    fun replaceLines(lines: Collection<Line>) {
+    fun clearLines() {
         if (status != Status.Fetching) return
         unfiltered.clear()
         filtered.clear()
-        unfiltered.addAll(lines)
-        for (line in lines) {
-            if (line.isVisible) filtered.add(line)
+    }
+
+    private fun addDateLine(array: ArrayDeque<Line>, newTimestamp: Long) {
```

Comment:
this runs at every line addition and might be simplified a bit? also functions that might or might not produce side effects better reflect that in the name, e.g. `addDateChangeLineIfDateChanged()`

Reply:
what exactly do you want to simplify? The name change sounds good to me, will change later.

Reply:
this code would run for every line addition in background. it creates and discards 4 date(?) objects for every addition. maybe this would be a case of premature optimization but i like to keep the code that might affect battery life as slim as possible, at least the new parts of it

Reply:
> this code would run for every line addition in background. it creates and discards 4 date(?) objects for every addition. 

I only count two (oldDate and newDate). I could cache the oldDate if you want.